packages/transformer/src/IModelTransformer.ts

```diff
@@ -727,6 +717,30 @@ export class IModelTransformer extends IModelExportHandler {
     /* eslint-enable @itwin/no-internal */
   }
 
+  /** validates that the importer set on the transformer has the same values for its shared options as the transformer.
+   *  @note This expects that the importer is already set on the transformer.
+   */
+  private validateSharedOptionsMatch() {
+    if (
+      Boolean(this._options.preserveElementIdsForFiltering) !==
+      this.importer.options.preserveElementIdsForFiltering
+    ) {
+      const errMessage =
+        "A custom importer was passed as a target but its 'preserveElementIdsForFiltering' option is out of sync with the transformer's option.";
+      Logger.logError(loggerCategory, errMessage);
```

Comment:
don't log what you throw, let the consumer decide if they want to log it

Reply:
not sure if historically I've followed that enough myself, but I think it's best practice